src/librekey/rnp_key_store.cpp

```diff
@@ -499,7 +505,7 @@ rnp_key_store_add_key(rnp_key_store_t *keyring, pgp_key_t *srckey)
         try {
             keyring->keys.emplace_back();
             added_key = &keyring->keys.back();
-            keyring->keybyfp[pgp_key_get_fp(srckey)] = std::prev(keyring->keys.end());
+            keyring->keybyfp[srckey->fp()] = std::prev(keyring->keys.end());
```

Comment:
Does this mean, if adding a mapping causes allocation exception, the `keyring->keys` will have an uninitialized element at the end?